go/vt/vtctl/grpcvtctldserver/testutil/test_backupstorage.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testutil
+
+import (
+	"context"
+	"sort"
+
+	"vitess.io/vitess/go/vt/mysqlctl/backupstorage"
+)
+
+type backupStorage struct {
+	backupstorage.BackupStorage
+
+	// Backups is a mapping of directory to list of backup names stored in that
+	// directory.
+	Backups map[string][]string
+	// ListBackupsError is returned from ListBackups when it is non-nil.
+	ListBackupsError error
+}
+
+// ListBackups is part of the backupstorage.BackupStorage interface.
+func (bs *backupStorage) ListBackups(ctx context.Context, dir string) ([]backupstorage.BackupHandle, error) {
```

Comment:
Implementing a whole new fake is so much work but it will definitely pay off 🏆 thank you!